cognite/_utils.py

```diff
@@ -79,12 +95,46 @@ def post_request(url, body, headers=None, params=None, cookies=None, use_gzip=Fa
     """Perform a POST request with a predetermined number of retries."""
     _log_request("POST", url, body=body, params=params, headers=headers, cookies=cookies)
 
-    data = json.dumps(body)
-    headers = headers or {}
-    if use_gzip:
-        headers["Content-Encoding"] = "gzip"
-        data = gzip.compress(json.dumps(body).encode("utf-8"))
-    return requests.post(url, data=data, headers=headers, params=params, cookies=cookies)
```

Comment:
Use what is on the master branch here. Retries and exceptions are handled with the `request_method` decorator, and the gzip stuff is simpler on master.

Reply:
Just add the `default=serialize` flag

Reply:
Reverted to what was in master: https://github.com/cognitedata/cognite-sdk-python/pull/150/commits/2876a5c8f524d4a0fcf2ccff6efe7cdd1bca2765
Added default serializer: https://github.com/cognitedata/cognite-sdk-python/pull/150/commits/74294632eedfb7fca8613472a690a7f7f6164be0